### PR TITLE
Design doc updates

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9646)

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -570,41 +570,25 @@ timestamps such thats<sub>1</sub> \< s<sub>2</sub>.
...
-pairs covering accounting, range metadata and node accounting
-before the actual key / value pairs for non-system data
-(e.g. the actual meat of the map).
+Logically, the map contains a series of reserved system key/value
+pairs preceding the actual user data (which is managed by the SQL
+subsystem).
 
-- `\0\0meta1` Range metadata for location of `\0\0meta2`.
-- `\0\0meta1<key1>` Range metadata for location of `\0\0meta2<key1>`.
+- `\x02<key1>`: Range metadata for range ending `\x03<key1>`.
```
Comment (petermattis) [inline docs/design.md]: I think it is worth mentioning this is a `meta1` key. Ditto for the `\x03` and `\x04` prefix below.

File: docs/design.md [hunk 2]
```diff
@@ -570,41 +570,25 @@ timestamps such thats<sub>1</sub> \< s<sub>2</sub>.
...
-- ...
-- `\0zone<keyN>`: Zone information for key prefix keyN.
-- `<>acctd<metric0>`: Accounting data for Metric 0 for empty key prefix.
-- ...
-- `<>acctd<metricN>`: Accounting data for Metric N for empty key prefix.
+- `\x03<keyN>`: Range metadata for range ending `<keyN>`.
+- `\x04{desc,node,range,store}-idegen`: ID generation oracles for various component types.
+- `\x04status-node-<varint encoded Store ID>`: Store runtime metadata.
+- `\x04tsd<...>`: Time-series data.
 - `<key0>`: `<value0>` The first user data key.**
```
Comment (petermattis) [inline docs/design.md]: `user data key` is mildly confusing. Perhaps `table data key`.

**Review by petermattis — APPROVED** [review]
LGTM

File: docs/design.md [hunk 3]
```diff
@@ -209,27 +209,27 @@ on a node is at a timestamp < next HLC time.
 
 Transactions are executed in two phases:
 
-1. Start the transaction by writing a new entry to the system
-   transaction table (keys prefixed by *\0tx*) with state “PENDING”. In
+1. Start the transaction by selecting a range which is likely to be
```
Comment (spencerkimball) [inline docs/design.md]: s/is likely to be heavily involved in the transaction/involved in the first write of the transaction/

**Review by spencerkimball — APPROVED** [review]

File: docs/design.md (hunk 3 above)
Comment (tamird) [inline docs/design.md]: Why? Isn't this what we just discussed as being the more-likely-to-rot variant?

Comment (tschottdorf) [inline docs/design.md]: Yeah, agree that this is too specific.

File: docs/design.md [hunk 4]
```diff
@@ -267,7 +267,7 @@ encounter data that necessitate conflict resolution
 
 When a transaction restarts, it changes its priority and/or moves its
 timestamp forward depending on data tied to the conflict, and
-begins anew reusing the same tx id. The prior run of the transaction might
+begins anew reusing the same txn id. The prior run of the transaction might
 have written some write intents, which need to be deleted before the
 transaction commits, so as to not be included as part of the transaction.
```
Comment (tschottdorf) [inline docs/design.md]: Not correct, epochs take care of this.

Comment (tschottdorf) [inline docs/design.md]: The whole section from here down is bogus.

File: docs/design.md [hunk 5]
```diff
@@ -281,12 +281,12 @@ a NOOP.
...
 
 This is the case in which a transaction, upon reading its transaction
-table entry, finds that it has been aborted. In this case, the
-transaction can not reuse its intents; it returns control to the client
-before cleaning them up (other readers and writers would clean up
-dangling intents as they encounter them) but will make an effort to
-clean up after itself. The next attempt (if applicable) then runs as a
-new transaction with **a new tx id**.
+record, finds that it has been aborted. In this case, the transaction
+can not reuse its intents; it returns control to the client before
```
Comment (tschottdorf) [inline docs/design.md]: It never really reuses intents. The intents' primary role is that of preventing starvation by forcing conflicts.

File: docs/design.md [hunk 6]
```diff
@@ -281,12 +281,12 @@ a NOOP.
 ***Transaction abort:***
 
 This is the case in which a transaction, upon reading its transaction
-table entry, finds that it has been aborted. In this case, the
-transaction can not reuse its intents; it returns control to the client
-before cleaning them up (other readers and writers would clean up
-dangling intents as they encounter them) but will make an effort to
-clean up after itself. The next attempt (if applicable) then runs as a
-new transaction with **a new tx id**.
+record, finds that it has been aborted. In this case, the transaction
```
Comment (tschottdorf) [inline docs/design.md]: s/upon/usually upon/ (because abort span, etc)

File: docs/design.md [hunk 7]
```diff
@@ -570,41 +570,21 @@ timestamps such thats<sub>1</sub> \< s<sub>2</sub>.
...
-- ...
-- `<keyN>`: `<valueN>` The last user data key.**
-
-There are some additional system entries sprinkled amongst the
-non-system keys. See the Key-Prefix Accounting section in this document
-for further details.
+- `\x03<keyN>`: Range metadata for range ending `<keyN>`. This a "meta2" key.
+- `\x04{desc,node,range,store}-idegen`: ID generation oracles for various component types.
+- `\x04status-node-<varint encoded Store ID>`: Store runtime metadata.
+- `\x04tsd<key>`: Time-series data key.
```
Comment (tschottdorf) [inline docs/design.md]: Where did zones go? Are they in one of the magic tables?

**Comment by tamird** [comment]
@tschottdorf can the lies which I re-wrapped be addressed subsequently? Zones are now in a SQL table.

**Comment by tschottdorf** [comment]
Sure.

**Pull request merged**